tasks/configure.yml

```diff
@@ -8,17 +16,48 @@
     - "{{monasca_agent_check_plugin_dir}}"
     - "{{monasca_agent_detection_plugin_dir}}"
 
-- name: Create additional plugins config
-  template: dest="{{monasca_conf_dir}}/agent/conf.d/{{item.key}}.yaml" src=plugin.yaml.j2 owner=root group=root mode=644
+- name: Run monasca-setup to create all default config files
+  command: "{{ agent_reconfigure_script }}"
+
+- name: Check for existing plugins configs created by monasca-setup
+  stat: path="{{ monasca_conf_dir }}/agent/conf.d/{{ item.key }}.yaml"
+  register: confs
+  ignore_errors: yes
   with_dict: monasca_checks
-  notify: run monasca-setup
 
-# Instead of running this directly by creating a file to run it changes such as user/pass will trigger a rerun. Also a user can run it manually
-- name: Create reconfigure script
-  template: dest="{{agent_reconfigure_script}}" src=monasca-reconfigure.j2 owner=root group=root mode=750
-  notify: run monasca-setup
+- name: Create additional plugins config
```

Comment:
There is some rather complicated reworking of the additional configuration to fix things up. Looking at it a bit the only problem seems to be that the '--overwrite' flag is specified in the monasca-reconfigure script and it shouldn't be in the case of any additional configuration being specified. Wouldn't fixing that solve the problems with additional config and leave the merging logic to monasca-setup?